### ISSUE TITLE
Questions about reproducibility: FF++ preprocessing, detector checkpoints, and generated dataset

### ISSUE THREAD

**Issue by willpower057** [laiyingxin2/Agent4FaceForgery#2]
Thank you for releasing the code for **Agent4FaceForgery**. I am currently studying the paper and trying to understand/reproduce the reported experiments.

After going through the released code, I have a few questions regarding the preprocessing and downstream detector experiments.

### 1. Preprocessing for the Phase 1 Xception judge vs. the final forgery detector

In [`[dataio/ffpp.py](https://github.com/laiyingxin2/Agent4FaceForgery/blob/main/dataio/ffpp.py)`](https://github.com/laiyingxin2/Agent4FaceForgery/blob/main/dataio/ffpp.py), the code appears to operate on **pre-extracted FF++ frames** stored on disk. `iter_real_frames()` directly loads the image with `PIL.Image.open()` without an explicit face detection/cropping step.

Similarly, the Xception training code appears to directly use these frames as input to the Xception judge.

Could you clarify whether the intended pipeline is:

```text
FF++ video
    ↓
pre-extracted frame
    ↓
Xception (Phase 1 ARS judge)
```

without RetinaFace-based face cropping?

In contrast, the paper states:

> "We use LLaVA [29], as the backbone model, and RetinaFace [9] to detect facial areas and scaled the face image to 224 × 224 with a patch size of 16."

So is the intended distinction that:

```text
Phase 1 ARS:
FF++ frame → Xception judge

Final detector:
FF++ / generated image
→ RetinaFace face detection
→ face crop
→ 224×224
→ LLaVA
```

If so, could you clarify where the RetinaFace preprocessing for the final LLaVA detector is implemented?

Also, is the same RetinaFace-based preprocessing applied consistently to the generated Agent4FaceForgery data and the cross-dataset evaluation sets?

### 2. Which FF++ compression setting was used?

The released `ffpp.py` appears to support the three standard FF++ compression settings:

* `raw`
* `c23`
* `c40`

Could you clarify which version was actually used for the experiments reported in the paper?

In particular, was the experiment based on:

* the original FF++ `raw` videos,
* FF++ `c23`,
* FF++ `c40`, or
* a pre-extracted frame version of one of these (e.g., the commonly distributed 200K-frame FF++/c23 subset)?

This would be very helpful because the compression level can have a significant effect on face-forgery detection performance and on the ARS judge.

### 3. Detector checkpoints and training code

For reproducibility of the main experimental results, would it be possible to provide the detector code/configuration and pretrained checkpoints for at least the following two settings?

**A. Baseline detector**

```text
FF++
  ↓
Detector
```

**B. Detector trained with the proposed data**

```text
FF++ + Agent4FaceForgery dataset
  ↓
Detector
```

Ideally, I would appreciate:

* the model/training code,
* pretrained checkpoints,
* training configuration/hyperparameters,
* preprocessing configuration,
* and the exact evaluation protocol.

In particular, since the paper uses LLaVA as the backbone for the final detector, it would be helpful to know the exact LLaVA checkpoint/version and fine-tuning configuration used for the reported results.

The current repository provides `to_llava_sft.py` to convert the generated data into an LLaVA SFT format, but I could not find the corresponding detector training script/checkpoint used to obtain the reported results.

### 4. Release of the generated dataset

I understand that the generated dataset may be quite large, and that releasing the complete image dataset may not be convenient.

Would it be possible to release either:

* the full Agent4FaceForgery-generated dataset, or
* a representative subset sufficient to reproduce/debug the training pipeline?

Even a smaller subset containing the images, labels, captions, and metadata would be very useful.

For example, a minimal reproducibility package containing:

```text
sample generated dataset
+
FF++ baseline detector checkpoint
+
FF++ + Agent4FaceForgery detector checkpoint
+
training/evaluation code
```

would already make it much easier to verify the main claim of the paper.

Thank you again for releasing Agent4FaceForgery. I think these additional artifacts would be very helpful for researchers who would like to reproduce and build upon the work.
